bin/phyx2owl.js

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+const phyx = require('..');
+
+/*
+ * An application for converting input Phyx files to OWL ontologies.
+ */
+
+// Read command line arguments.
+const argv = require('yargs')
+  .usage("$0 [files or directories to convert into OWL ontologies]")
+  .describe('max-internal-specifiers', 'The maximum number of internal specifiers (phylorefs with more than this number will be ignored)')
+  .default('max-internal-specifiers', 8)
+  .describe('max-external-specifiers', 'The maximum number of external specifiers (phylorefs with more than this number will be ignored)')
+  .default('max-external-specifiers', 8)
+  .help()
+  .alias('h', 'help')
+  .argv
+
+/*
+ * Get a list of all files in a directory. We will recurse into directories and choose
+ * files that meet the criteria in the function `check(filename) => boolean`.
+ */
+function getFilesInDir(filePath, check = (filename => filename.toLowerCase().endsWith(".json"))) {
+  // console.debug(`Processing file: ${dir}`)
+  if (!fs.existsSync(filePath)) return [];
+
+  const lsync = fs.lstatSync(filePath);
+  if (lsync.isFile()) {
+    // If `path` is a file, check if it meets the provided requirement. If so,
+    // add it to the list of collected files.
+    if (!check(filePath)) {
+      // console.log(`Skipping ${dir}.`)
+      return [];
+    } else {
+      return [filePath];
+    }
+  } else if (lsync.isDirectory()) {
+    // If `path` is a directory, recurse into every file in that directory.
+    const files = fs.readdirSync(filePath);
+    return files.map(file => getFilesInDir(path.join(filePath, file), check))
+      .reduce((acc, curr) => acc.concat(curr), [])
+      .filter(filename => filename);
```

Comment:
What's the point of this `filter()`?

Reply:
It's intended to be a test for truthyness, although apparently a better way of writing that is `.filter(Boolean)`. However, I don't think this is needed here, since the output from `getFilesInDir()` is already a list of lists that needs to be reduced before it can be used as a single list, so any missing entries should just be empty lists already. I've removed this filter in 24526bf9912 and it seems to work okay.